[WIP] fix Travis CI

peterbecich opened the pull request:
https://github.com/haskell/haskell-ide-engine/issues/1229

This is close to having Travis CI working for both Ubuntu and OS X.
I've removed the `deploy` stage; figured it had been replaced by now.  Should it remain?

peterbecich committed:
WIP fix Travis CI

Squashed commit of the following:

commit 2275fa60537279f5f519ee368731a126492f2117
Author: Peter Becich <peterbecich@gmail.com>
Date:   Sat Nov 16 13:58:50 2019 -0800

    clean-up

commit c67eafd7dff7b8c454be81bedc5a55919d80ecda
Author: Peter Becich <peter@simspace.com>
Date:   Tue Nov 12 20:13:56 2019 -0800

    lens

commit 94879f17096816ca7fd0dcae434828dc1211faeb
Author: Peter Becich <peterbecich@gmail.com>
Date:   Mon Nov 11 23:34:56 2019 -0800

    travis Ubuntu and OS X

commit 2cf53fd971e894945afbdc012c87f24b8c0aab3d
Author: Peter Becich <peterbecich@gmail.com>
Date:   Mon Nov 11 22:43:37 2019 -0800

    build, test stages

commit f94572930043e7b6e4ff9fba4a5e7c8981a657af
Author: Peter Becich <peterbecich@gmail.com>
Date:   Mon Nov 11 20:04:17 2019 -0800

    use Ubuntu for Travis

commit b7ccd2f349ba92d009162033285670caf32217b4
Author: Peter Becich <peterbecich@gmail.com>
Date:   Mon Nov 11 17:32:20 2019 -0800

    test more GHC versions on Travis

commit d2e4f10d3c78b2d7ebaadcb80f89d9852c9c5017
Author: Peter Becich <peterbecich@gmail.com>
Date:   Mon Nov 11 16:37:57 2019 -0800

    fix travis

bubba commented:
Thanks! Looks like travis CI isn't running for PRs. I think we need a Haskell org admin to turn it on?

peterbecich committed:
Travis: two stages for dependencies

peterbecich commented:
I'm expecting this to pass: https://travis-ci.org/peterbecich/haskell-ide-engine/builds/613034877

mpickering commented:
Why do we need travis, circle, azure and github actions CI?

bubba commented:
GitHub actions is just experimental, don’t consider it. CircleCI was needed for Linux, Azure was needed for windows, Travis was needed for Mac. Now they all support all operating systems though. 

> On 17 Nov 2019, at 09:22, Matthew Pickering <notifications@github.com> wrote:
> 
> ﻿
> Why do we need travis, circle, azure and github actions CI?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.

peterbecich committed:
travis ci retry

lorenzo commented:
The idea of having Travis was to generate binaries for all platforms after each tag. That was before discovering that the binaries we produce can only work in the machine they were compiled at.

bubba commented on .travis.yml:
```diff
@@ -22,104 +27,521 @@ cache:
+     os: osx
+     osx_image: xcode11.2
+     env: GHC_VER="8.6.5"
+     script: *test
```
Is there a way to specify a matrix for the GHC_VER env?

peterbecich replied:
I don't think so.  Would have been nice.  Seems that stages and a matrix of environment variables cannot be used together:
https://github.com/peterbecich/haskell-ide-engine/blob/aadbea8eb9d1212dc3752ac5a843469d5f42cd94/.travis.yml
https://travis-ci.org/peterbecich/haskell-ide-engine/builds/618340235
I don't know why the Test stage expands differently than the other stages in this build.

To me this documentation implies it would work for something other than an environment variable: https://docs.travis-ci.com/user/build-stages/#build-stages-and-build-matrix-expansion

peterbecich commented:
I'm fixing the broken Test stage now

peterbecich commented on .travis.yml:
```diff
@@ -22,104 +27,521 @@ cache:
   - $TRAVIS_BUILD_DIR/submodules/cabal-helper/.stack-work
   - $TRAVIS_BUILD_DIR/hie-plugin-api/.stack-work
-  timeout: 800
+  timeout: 2000
```
seconds https://docs.travis-ci.com/user/caching/#setting-the-timeout

peterbecich committed:
Merge branch 'master' of github.com:haskell/haskell-ide-engine into fix-travis-ci

peterbecich committed:
attempt to fix Travis test stage

Squashed commit of the following:

commit 84cac4a9f75b3d0b022851b8a5510952168fdb95
Merge: 31f6bad5 ae935f43
Author: Peter Becich <peterbecich@gmail.com>
Date:   Fri Nov 29 15:48:29 2019 -0800

    Merge branch 'master' of github.com:haskell/haskell-ide-engine into travis-stack-test-2

commit 31f6bad5777fb33b1b3e9a5d25080fbca65825f6
Author: Peter Becich <peterbecich@gmail.com>
Date:   Fri Nov 29 00:32:49 2019 -0800

    more test dependencies

commit 479bd7bdb6bc9f076223dee4d7eaeefcf4554345
Author: Peter Becich <peterbecich@gmail.com>
Date:   Thu Nov 28 11:37:13 2019 -0800

    test dependencies

commit 7cad32bc8086e047674990dd7415ef02113c6f40
Author: Peter Becich <peterbecich@gmail.com>
Date:   Wed Nov 27 17:17:49 2019 -0800

    fixup! fixup! fixup! fixup! fixup! Merge branch 'master' into travis-stack-test

commit ceeb03deffd1a45056fb32fa0feafe28a327c0c3
Author: Peter Becich <peterbecich@gmail.com>
Date:   Tue Nov 26 21:51:40 2019 -0800

    fixup! fixup! fixup! fixup! Merge branch 'master' into travis-stack-test

commit 45cff227b2a9682f324fb26bf1ecff5c76023f56
Author: Peter Becich <peterbecich@gmail.com>
Date:   Mon Nov 25 19:37:10 2019 -0800

    fixup! fixup! fixup! Merge branch 'master' into travis-stack-test

commit 2fa038e586afb2709084f76bf4c2745a40c95340
Author: Peter Becich <peterbecich@gmail.com>
Date:   Sun Nov 24 12:20:57 2019 -0800

    fixup! fixup! Merge branch 'master' into travis-stack-test

commit 904113ec304d4c9a3f6d18dd50e25fd66767d93e
Author: Peter Becich <peterbecich@gmail.com>
Date:   Sun Nov 24 12:09:54 2019 -0800

    fixup! Merge branch 'master' into travis-stack-test

commit 5fd226ad6b3565c9efa2b59e83eb1e21a3651c79
Merge: 3325d2f8 c1e57f5c
Author: Peter Becich <peterbecich@gmail.com>
Date:   Sat Nov 23 21:17:08 2019 -0800

    Merge branch 'master' into travis-stack-test

commit 3325d2f8a0e94bb0e9101a46be153791d9ab9658
Author: Peter Becich <peterbecich@gmail.com>
Date:   Sat Nov 23 03:03:40 2019 -0800

    travis stack test

peterbecich committed:
verbose test
